src/main/groovy/com/twcable/grabbit/jcr/ProtoNodeDecorator.groovy

```diff
@@ -97,6 +97,10 @@ class ProtoNodeDecorator {
                 log.debug "Added mixin ${value.stringValue} for : ${node.name}."
             }
             else {
+                JcrNodeDecorator decoratedNode = new JcrNodeDecorator(node)
```

Comment:
I think these are details that ProtoNodeDecorator, or any other consumer probably shouldn't know about. ProtoNodeDecorator just wants to write a mixin in this case; It shouldn't need to know the details. 

To extend that thought, I think we are doing too much within writeToJcr. We should hand JcrNodeDecorator a ProtoNodeDecorator, and it should have all the information needed to build itself.